xdsl/irdl/irdl.py

```diff
@@ -1074,6 +1095,18 @@ class OpDef:
     """
     assembly_format: str | None = field(default=None)
 
+    @property
+    def traits(self) -> frozenset[OpTrait]:
+        if callable(self._traits):
+            self._traits = self._traits()
+        return self._traits
+
+    @traits.setter
+    def traits(
+        self, traits: frozenset[OpTrait] | Callable[[], frozenset[OpTrait]]
+    ) -> None:
+        self._traits = traits
```

Comment:
Do we really want a world-accessible setter for traits? I don't think these should be modifiable outside of IRDL, no?

Reply:
I don't see the disadvantage

Reply:
I believe we need it, mostly for modularity reason.
If someone is defining a new trait, and they want to apply it to existing dialects, they have to override this trait here ;)

Reply:
Scary business